src/lib/Internal/TypeInspection.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Reflection;
+
+namespace Ockham.Data
+{
+
+    /// <summary>
+    /// NOT part of the public API of this project. Type inspection per se is not the goal of the Ockham.NET.Convert module
+    /// </summary>
+    internal class TypeInspection
```

Comment:
Note this class is internal and does not affect the public API as approved in #2, #3